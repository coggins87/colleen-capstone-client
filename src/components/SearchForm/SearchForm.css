.check_box_list {
  list-style-type: none;
  column-count: 2;
}
.check_box_list > li {
  white-space: nowrap;
}

.red {
  background-color: red;
  font-weight: bold;
  text-align: center;
  padding: 10px;
  border: 2px solid red;
  margin-bottom: 2%;
}

@media all and (max-width: 509px) {
  .search_form_inputs {
    display: flex;
    justify-content: center;
  }
  .check_box_list {
    column-count: 1;
  }
  .search_form {
    padding-bottom: 15%;
  }
}

.equipment_field {
  margin-top: 3%;
  border: none;
  margin-bottom: -3%;
}

.equipment_select {
  display: flex;
}

.time_label,
.time_input {
  font-size: 0.95em;
}

.search_form > .form_button {
  align-content: space-around;
}
