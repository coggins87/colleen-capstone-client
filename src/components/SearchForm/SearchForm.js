import React from 'react'
//import CheckBoxBody from '../FormHelpers/CheckBoxBody'
import CheckBoxEquip from '../FormHelpers/CheckBoxEquip'
import Result from '../../components/Results/results'
import WorkoutSearchService from '../../services/workout-search-service'
class SearchForm extends React.Component {
  constructor(props){
    super(props)
    
    this.state = {
      error: null,
      time: 0,
      equipment:[
      {id: 7, value: "dumbell", isChecked: false},
      {id: 8, value: "barbell", isChecked: false},
      {id: 9, value: "treadmill", isChecked: false},
      {id: 10, value: "pullup-bar", isChecked: false},
      {id: 11, value: "medicine-ball", isChecked: false},
      {id: 12, value: "stationary-bike", isChecked: false},
      {id: 13, value: "bands", isChecked: false}],
      result: [],
    }
  }
  
  updateTime = e => {
    console.log(' SET STATE ')

    this.setState({
      time: e.target.value
    })
  }
/*   updateBodyParts = event => {
    let newParts = this.state.body_parts
    newParts.forEach(part => {
      if(part.value === event.target.value)
      part.isChecked = event.target.checked
    })
    this.setState({
      body_parts: newParts
    })
  }
 */
  updateEquipment = event => {
    let newEquipment = this.state.equipment
    newEquipment.forEach(equipment => {
      if(equipment.value === event.target.value)
      equipment.isChecked = event.target.checked
    })
    console.log(' SET STATE ')

    this.setState({
      
      equipment: newEquipment
    })
  }

  saveWorkout = e => {
    e.preventDefault()
    let time = this.state.time
    console.log(time)
    let userId = this.props.req.params.userId
    console.log(userId)
  }

  handleSubmit = e => {
    e.preventDefault()
   /*  let parts=[]
    let bodyParts = this.state.body_parts.filter(part => part.isChecked);
    bodyParts.forEach(item => {
      parts.push(item.value)
    }); */
    
    let time = this.state.time
    let equipment=[]
    let equipObj = this.state.equipment.filter(equip => equip.isChecked);
    equipObj.forEach(item => {
      equipment.push(item.value)
    });
    let equipmentString = equipment.join(" ")

    WorkoutSearchService.submitSearch(time, equipmentString)
    .then(res=> {
      console.log('RESULTS SET STATE')
      this.setState({result: res})
      })
  
  }
  render(){
    
    return (    <>
      <form onSubmit={this.handleSubmit}>
        <label htmlFor="time">Select Your Workout Time Between 5 and 60 Minutes(required):</label>
        <input id="time" type='number' min='5' max='60' onChange={this.updateTime} required></input>
       {/*  <fieldset>
        <legend htmlFor="body_parts">Select Up to 5 Body Parts to Target (optional):</legend>
        <ul>
        {
          this.state.body_parts.map((part) => {
            return (<CheckBoxBody handleCheck={this.updateBodyParts}  {...part} />)
          })
        }
        </ul>

      </fieldset> */}
      <fieldset>
        <legend>Select Equipment (optional):</legend>
        <ul>
        {
          this.state.equipment.map((equipment) => {
            return (<CheckBoxEquip handleCheck={this.updateEquipment}  {...equipment} />)
          })
        }
        </ul>
        </fieldset>
        <button>Make My Workout</button>
      </form>
      {this.state.result.length > 0 ?  <Result result={this.state.result} time={this.state.time} saveWorkout={this.saveWorkout} /> : null}

      </>

    )
  }
}


export default SearchForm