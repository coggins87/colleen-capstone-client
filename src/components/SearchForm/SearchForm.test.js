import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SearchForm from './SearchForm'

jest.mock('uuid', () => jest.fn(() => '1'));

describe(`SearchForm component`, () => {
  const props = {
    testObj: {},
    history : {
    testCheck: ()=>{}
    },
  }

  it('renders a SearchForm by default', () => {
    const wrapper = shallow(<SearchForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the SearchForm given props', () => {
    const wrapper = shallow(<SearchForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

 
})