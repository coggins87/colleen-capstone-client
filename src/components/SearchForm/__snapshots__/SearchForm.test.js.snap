// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component renders a SearchForm by default 1`] = `
<Fragment>
  <div
    className="search_form_inputs"
  >
    <div
      role="alert"
    />
    <button
      id="toggle_form"
      onClick={[Function]}
    >
      -
    </button>
    <form
      className="_form search_form animated fadeIn"
      onSubmit={[Function]}
    >
      <label
        className="time_label"
        htmlFor="time"
      >
        Select Workout Time 5-60 Minutes (required):
         
      </label>
      <input
        className="time_input"
        id="time"
        max="60"
        min="5"
        onChange={[Function]}
        required={true}
        type="number"
      />
      <div
        className="equiment_select"
      >
        <fieldset
          className="equipment_field"
        >
          <legend>
            Select Equipment (optional):
          </legend>
          <ul
            className="check_box_list"
          >
            <li
              key="1"
            >
              <CheckBoxEquip
                handleCheck={[Function]}
                id={7}
                isChecked={false}
                value="dumbell"
              />
            </li>
            <li
              key="1"
            >
              <CheckBoxEquip
                handleCheck={[Function]}
                id={8}
                isChecked={false}
                value="barbell"
              />
            </li>
            <li
              key="1"
            >
              <CheckBoxEquip
                handleCheck={[Function]}
                id={9}
                isChecked={false}
                value="treadmill"
              />
            </li>
            <li
              key="1"
            >
              <CheckBoxEquip
                handleCheck={[Function]}
                id={10}
                isChecked={false}
                value="pullup-bar"
              />
            </li>
            <li
              key="1"
            >
              <CheckBoxEquip
                handleCheck={[Function]}
                id={11}
                isChecked={false}
                value="medicine-ball"
              />
            </li>
            <li
              key="1"
            >
              <CheckBoxEquip
                handleCheck={[Function]}
                id={12}
                isChecked={false}
                value="stationary-bike"
              />
            </li>
            <li
              key="1"
            >
              <CheckBoxEquip
                handleCheck={[Function]}
                id={13}
                isChecked={false}
                value="bands"
              />
            </li>
          </ul>
        </fieldset>
      </div>
      <button
        className="form_button"
      >
        Make My Workout
      </button>
    </form>
  </div>
</Fragment>
`;

exports[`SearchForm component renders the SearchForm given props 1`] = `
<Fragment>
  <div
    className="search_form_inputs"
  >
    <div
      role="alert"
    />
    <button
      id="toggle_form"
      onClick={[Function]}
    >
      -
    </button>
    <form
      className="_form search_form animated fadeIn"
      onSubmit={[Function]}
    >
      <label
        className="time_label"
        htmlFor="time"
      >
        Select Workout Time 5-60 Minutes (required):
         
      </label>
      <input
        className="time_input"
        id="time"
        max="60"
        min="5"
        onChange={[Function]}
        required={true}
        type="number"
      />
      <div
        className="equiment_select"
      >
        <fieldset
          className="equipment_field"
        >
          <legend>
            Select Equipment (optional):
          </legend>
          <ul
            className="check_box_list"
          >
            <li
              key="1"
            >
              <CheckBoxEquip
                handleCheck={[Function]}
                id={7}
                isChecked={false}
                value="dumbell"
              />
            </li>
            <li
              key="1"
            >
              <CheckBoxEquip
                handleCheck={[Function]}
                id={8}
                isChecked={false}
                value="barbell"
              />
            </li>
            <li
              key="1"
            >
              <CheckBoxEquip
                handleCheck={[Function]}
                id={9}
                isChecked={false}
                value="treadmill"
              />
            </li>
            <li
              key="1"
            >
              <CheckBoxEquip
                handleCheck={[Function]}
                id={10}
                isChecked={false}
                value="pullup-bar"
              />
            </li>
            <li
              key="1"
            >
              <CheckBoxEquip
                handleCheck={[Function]}
                id={11}
                isChecked={false}
                value="medicine-ball"
              />
            </li>
            <li
              key="1"
            >
              <CheckBoxEquip
                handleCheck={[Function]}
                id={12}
                isChecked={false}
                value="stationary-bike"
              />
            </li>
            <li
              key="1"
            >
              <CheckBoxEquip
                handleCheck={[Function]}
                id={13}
                isChecked={false}
                value="bands"
              />
            </li>
          </ul>
        </fieldset>
      </div>
      <button
        className="form_button"
      >
        Make My Workout
      </button>
    </form>
  </div>
</Fragment>
`;
